actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -1304,44 +1314,74 @@ private void updateKeyCloakUserBase(Map<String, Object> userMap) {
-    actorMessage.getRequest().putAll(userMap);
-    UserRequestValidator.validateCreateUser(actorMessage);
-    Util.checkPhoneUniqueness(userMap, JsonKey.CREATE);
-    Util.checkEmailUniqueness(userMap, JsonKey.CREATE);
-    Map<String, Object> emailTemplateMap = new HashMap<>(userMap);
-    if (userMap.containsKey(JsonKey.WEB_PAGES)) {
-      SocialMediaType.validateSocialMedia(
-          (List<Map<String, String>>) userMap.get(JsonKey.WEB_PAGES));
-    }
-    userMap.put(JsonKey.CREATED_BY, req.get(JsonKey.REQUESTED_BY));
     // remove these fields from req
     userMap.remove(JsonKey.ENC_EMAIL);
     userMap.remove(JsonKey.ENC_PHONE);
     userMap.remove(JsonKey.EMAIL_VERIFIED);
-    // Will ignore these fields in create api
-    userMap.remove(JsonKey.EXTERNAL_ID);
-    userMap.remove(JsonKey.PROVIDER);
-    userMap.remove(JsonKey.EXTERNAL_ID_PROVIDER);
-    userMap.remove(JsonKey.ID_TYPE);
-    userMap.remove(JsonKey.EXTERNAL_ID_TYPE);
+    userMap.put(JsonKey.CREATED_BY, req.get(JsonKey.REQUESTED_BY));
+    actorMessage.getRequest().putAll(userMap);
     try {
+      String channel = (String) userMap.get(JsonKey.CHANNEL);
       // validate channel and set rootOrgId of user
-      if (StringUtils.isBlank((String) userMap.get(JsonKey.CHANNEL))) {
-        userMap.put(JsonKey.CHANNEL, ProjectUtil.getConfigValue(JsonKey.SUNBIRD_DEFAULT_CHANNEL));
+      if (StringUtils.isBlank(channel)) {
+        channel = ProjectUtil.getConfigValue(JsonKey.SUNBIRD_DEFAULT_CHANNEL);
+        userMap.put(JsonKey.CHANNEL, channel);
       }
-      String rootOrgId = Util.getRootOrgIdFromChannel((String) userMap.get(JsonKey.CHANNEL));
+      String rootOrgId = Util.getRootOrgIdFromChannel(channel);
       userMap.put(JsonKey.ROOT_ORG_ID, rootOrgId);
     } catch (Exception ex) {
       sender().tell(ex, self());
       return;
     }
+    String version = (String) actorMessage.getRequest().get(JsonKey.VERSION);
+    if (StringUtils.isNotBlank(version) && "V2".equalsIgnoreCase(version)) {
+      UserRequestValidator.validateCreateUserV2(actorMessage);
+      validateChannelAndOrganisationId(userMap);
+    } else {
+      // For V1
+      UserRequestValidator.fieldsNotAllowed(Arrays.asList(JsonKey.ORGANISATION_ID), actorMessage);
+      UserRequestValidator.validateCreateUser(actorMessage);
+    }
+    processUserRequest(userMap);
+  }
+
+  private void validateChannelAndOrganisationId(Map<String, Object> userMap) {
+    String organisationId = (String) userMap.get(JsonKey.ORGANISATION_ID);
+    String channel = (String) userMap.get(JsonKey.CHANNEL);
+    if (StringUtils.isNotBlank(organisationId)) {
+      Map<String, Object> orgMap = Util.getOrgDetails(organisationId);
+      if (MapUtils.isEmpty(orgMap)) {
+        ProjectCommonException.throwClientErrorException(ResponseCode.invalidOrgData, null);
+      }
+      String subOrgRootOrgId = (String) orgMap.get(JsonKey.ROOT_ORG_ID);
```

Comment:
Check isRootOrg and based on that fetch rootOrgId of sub-org and do the comparison.